external-crates/move/crates/move-analyzer/src/symbols/compilation.rs

```diff
@@ -428,23 +473,123 @@ pub fn get_compiled_pkg(
     Ok((Some(compiled_pkg_info), ide_diagnostics))
 }
 
-fn has_precompiled_deps(
-    pkg_path: &Path,
-    pkg_dependencies: Arc<Mutex<BTreeMap<PathBuf, PrecomputedPkgInfo>>>,
-) -> bool {
+/// Get compiled dependencies from cache or compile them if either
+/// - not present in the cache (updata the cache if version of dependencies requested
+///   is one with function bodies)
+/// - a fully-compiled (with function bodies) version of the dependency is requested
+///   (do not update the cache in this case)
+///
+/// The reason for it is that in steady state (after the first full compilation)
+/// only smaller version of deps are shared between packages.
```

Comment:
Why does this not retain the fully-compiled version? Or perhaps I misunderstand the comment?

Reply:
We only want the "light" version to be kept around. The "heavy" (fully compiled) version does not have to be kept around beyond the first compilation, and should not be to reduce the memory footprint. I may be misunderstanding the question, though...